### ISSUE TITLE
oob read in mdb-tables

### ISSUE THREAD

**Issue by Miladbr** [mdbtools/mdbtools#114]
```
$ ./mdb-tables c11.mdb 
offset 12935430144 is beyond EOF
offset 12935430144 is beyond EOF
offset 3311470313472 is beyond EOF
[1]    15653 segmentation fault  ./mdb-tables c11.mdb
```

**backtrace:**
```
#0  0x00000000004036fc in mdb_read_catalog (mdb=0x638b90, objtype=0xffffffff) at catalog.c:108
#1  0x00000000004028ad in main (argc=<optimized out>, argv=<optimized out>, argv@entry=0x7fffffffe458) at mdb-tables.c:139
#2  0x00007ffff771d830 in __libc_start_main (main=0x4023b0 <main>, argc=0x2, argv=0x7fffffffe458, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffe448) at ../csu/libc-start.c:291
#3  0x0000000000401ff9 in _start ()
```

**PoC:**
```
https://github.com/Miladbr/public-poc/blob/master/mdbtools/c11.mdb
```

**Comment by evanmiller** [comment]
Hi, thanks for the report. This segfault is confirmed to be fixed in my branch: https://github.com/evanmiller/mdbtools/tree/master

I've been doing a lot of work fixing OOB errors and hope to get the work merged in eventually.

**Comment by evanmiller** [comment]
Fix included in #162
